src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -141,21 +161,17 @@ export class EthereumBlockchainService implements BlockchainService {
   private _chainId: number | undefined
   private readonly network: string
   private readonly transactionTimeoutSecs: number
-  private readonly contract: Contract
   private readonly overrideGasConfig: boolean
-  private readonly gasLimit: number
-  private readonly useSmartContractAnchors: boolean
-  private readonly contractAddress: string
+  private readonly gasLimit: bigint
+  private readonly contractAddress: Address
 
-  constructor(config: Config, private readonly wallet: ethers.Wallet) {
-    this.useSmartContractAnchors = config.useSmartContractAnchors
+  constructor(config: Config, private readonly provider: PublicClient, private readonly wallet: WalletClient, private readonly walletAddress: Address) {
```

Comment:
It is safer IMO to not mix `private readonly` constructor properties with property declarations like on 161-166. It makes it complicated to see what is actually available. IMO.

Reply:
I agree, will adjust that.